src/main/java/org/atlasapi/equiv/generators/SherlockTitleSearchGenerator.java

```
@@ -275,4 +261,110 @@ private String getName() {
     public String toString() {
         return name;
     }
+
+    public static class YearFilterConfiguration{
+
+        private int tolerance = 0;
+        private boolean allowsNull = false;
+
+        public YearFilterConfiguration setTolerance(int tolerance){
+            this.tolerance = tolerance;
+            return this;
+        }
+
+        public YearFilterConfiguration setAllowNull(){
+            return setAllowNull(true);
+        }
+
+        public YearFilterConfiguration setAllowNull(boolean allowsNull){
+            this.allowsNull = allowsNull;
+            return this;
+        }
+
+        private boolean allowsNulls() {
+            return allowsNull;
+        }
+
+        private int getTolerance() {
+            return tolerance;
+        }
+    }
+
+    public static class SherlockTitleSearchGeneratorBuilder<T extends Content> {
+
+        SherlockSearchResolver searchResolver;
+        Class<? extends T> cls;
+        Iterable<Publisher> publishers;
+        Function<String, String> titleTransform = Functions.identity();
+        int searchLimit;
+        Score exactMatchScore;
+        Score partialMatchBound;
+        boolean excludeSelfPublisher;
```

Since these booleans are primitives, if someone forgets to use the appropriate `with` method then these will silently default to false. Either make these `Boolean` objects with null checking or specify the default to avoid ambiguity.